api/clients/v2/accountant.go

```diff
@@ -74,50 +73,76 @@ func (a *Accountant) BlobPaymentInfo(
 	quorumNumbers []uint8,
 	timestamp int64) (*big.Int, error) {
 
+	symbolUsage := meterer.SymbolsCharged(numSymbols, a.minNumSymbols)
+
+	// Always try to use reservation first
+	payment, err := a.ReservationUsage(symbolUsage, quorumNumbers, timestamp)
+	if err == nil {
+		return payment, nil
+	}
+
+	// Fall back to on-demand payment if reservation fails
+	return a.OnDemandUsage(symbolUsage, quorumNumbers)
+}
+
+// ReservationUsage attempts to use the reservation for the given request.
+// Returns (0, nil) if successful, or (nil, error) if reservation cannot be used.
+func (a *Accountant) ReservationUsage(
+	symbolUsage uint64,
+	quorumNumbers []uint8,
+	timestamp int64) (*big.Int, error) {
```

Comment:
Instead of returning (*big.Int, error), we could just return error right? Then in BlobPaymentInfo we just do 
```
	payment, err := a.ReservationUsage(symbolUsage, quorumNumbers, timestamp)
	if err == nil {
		return big.NewInt(0), nil
	}
```

Reply:
makes sense! [updated](https://github.com/Layr-Labs/eigenda/pull/1653/commits/510fe1a875d031d2a9ce022ad8751582719e0f85)